azurerm/internal/services/cdn/cdn_profile_resource.go

```diff
@@ -67,9 +52,40 @@ func resourceArmCdnProfile() *schema.Resource {
 			"tags": tags.Schema(),
 		},
 	}
+
+	return &schema.Resource{
+		Create: resourceArmCdnProfileCreate,
+		Read:   resourceArmCdnProfileRead,
+		Update: resourceArmCdnProfileUpdate,
+		Delete: resourceArmCdnProfileDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.CdnProfileID(id)
+			return err
+		}),
+
+		Schema: tmpResource.Schema,
```

Comment:
the schema needs to remain here - and be entirely duplicated for the migration; else if these differ in the future the original state migration schema is no longer valid - so unfortunately these cannot be reused and must be copy/pasted as a point-in-time reference to the current schema

Reply:
As this is a not a schema migration essentially, it is just the ID got migrated. I think it is fine to reuse the schema at this point.

If the schema got changed in the future, it should not impact the fact that the old ID format will be transitioned to the new one as long as the `schema_version` in state is 0. On the other hand, if the schema introduced a breaking change and we want to keep backward compatibility, then we can do a point-in-time copy at that time (like a copy-on-write style).

In this way, it is like that the version 0 schema keeps evolving until we decide to snapshot it in a fixed version at the time that we really want to do some schema level migration.

What do you think?

Reply:
can we re-inline this to match other resources too?